dashboard/models.py

```diff
@@ -54,23 +55,15 @@ def user_qset(cls, user, program=None):
         """
         query_params = dict(user=user)
         if program is not None:
-            query_params.update(dict(course_run__course__program=program))
+            query_params.update(dict(
+                course_run__course__program=program,
+            ))
+        if provider is not None:
+            query_params.update(dict(
+                course_run__courseware_backend=provider
+            ))
         return cls.objects.filter(**query_params)
```

Comment:
Take it or leave it: this would probably be a bit cleaner with multiple query = query.filter(...) lines instead of building up a dict for a single call.